tree/tree.go

```diff
@@ -0,0 +1,76 @@
+package tree
+
+import "fmt"
+
+// Tree holds the tree data
+type Tree struct {
+	rootNode Node
+}
+
+// NewTree returns a new instance of the Tree
+func NewTree() *Tree {
+	return &Tree{
+		rootNode: NewRootNode(),
+	}
+}
+
+func (t *Tree) Root() []byte {
+	return t.rootNode.GetHash()
+}
+
+func (t *Tree) Get(key []byte) ([]byte, bool) {
```

Comment:
The pattern we have elsewhere (namely the `database.Database` interface) is that `Get` returns `([]byte,error)` where the error is `database.ErrNotFound` if the key isn't present

Reply:
Yep - fixed that :)